src/models/builders/nextjs-builder.ts

```diff
@@ -1,190 +1,71 @@
     matchers: string,
     filepath: string
 }
 
 class NextjsBuilder extends Builder {
-    dirname: string = dirname(__filename);
-    functionsMap: Map<string, string> = new Map();
-    tmpFunctionsDir: string = join(tmpdir(), Math.random().toString(36).slice(2));
-    invalidFunctions: string[] = [];
     hydratedMiddleware: Map<string, HydratedEntry> = new Map();
     hydratedFunctions: Map<string, HydratedEntry> = new Map();
+    functionsMap: Map<string, string> = new Map();
+    dirname: string = dirname(__filename);
     middlewareEntries: any;
     functionsEntries: any;
+    manifestBuilderService = new ManifestBuilderService();
+    vercelService = new VercelService();
+    esbuild = esbuild;
+
 
     constructor(targetDir: string) {
         super(targetDir);
     }
 
-    async createVercelProjectConfig() {
-        console.log("Creating project config file ...");
-
-        try {
-            const projectConfigDir = '.vercel';
-            const projectConfigFilePath = `${projectConfigDir}/project.json`;
-
-            if (!existsSync(projectConfigFilePath)) {
-                if (!existsSync(projectConfigDir)) {
-                    mkdirSync(projectConfigDir);
-                }
-
-                await writeFile(projectConfigFilePath, '{"projectId":"_","orgId":"_","settings":{}}');
-            }
-        } catch (error) {
-            const message = `Error: ${error}`;
-            console.log(message)
-            throw Error(message)
-        }
-    }
-
-    async runVercelBuild() {
-        // https://vercel.com/docs/build-output-api/v3
-        console.log("Running initial build ...");
-
-        try {
-            await exec('npx --yes vercel@28.16.11 build --prod');
-        } catch (error) {
-            const message = `Error: ${error}`;
-            console.log(message)
-            throw Error(message)
-        }
-    }
-
-    loadVercelConfigs(): any {
-        console.log("Loading configs ...")
-
-        try {
-            const fileContent = readFileSync(".vercel/output/config.json", "utf8");
-            const config = JSON.parse(fileContent);
-
-            return config;
-        } catch (error) {
-            const message = `Error: ${error}`;
-            console.log(message)
-            throw Error(message)
-        }
-    }
-
-    async detectBuildedFunctions() {
-        console.log("Detecting builded functions ...")
-
-        let functionsExist = false;
-
+    detectBuildedFunctions() {
+        console.log("Detecting builded functions ...");
         try {
             const functionsDir = resolve(".vercel/output/functions");
-
-            await stat(functionsDir);
-
-            functionsExist = true;
-        } catch (error) {
-            if (!functionsExist) {
-                console.log("No functions detected")
-            } else {
-                console.log(error)
-            }
-
-            throw Error("Failed in detect builded functions.")
+            statSync(functionsDir);
```

Comment:
I'm not sure if this function is really needed. Anyway, as we already have this, maybe can me moved to vercel-service class.

Reply:
Moved to vercel service